x-pack/plugins/security_solution/server/lib/detection_engine/signals/source_fields_merging/utils/is_ignored.test.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isIgnored } from './is_ignored';
+
+describe('is_ignored', () => {
+  beforeAll(() => {
+    jest.resetAllMocks();
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('string matching', () => {
+    test('it returns false if given an empty array', () => {
+      expect(isIgnored('simple.value', [])).toEqual(false);
+    });
+
+    test('it returns true if a simple string value matches', () => {
+      expect(isIgnored('simple.value', ['simple.value'])).toEqual(true);
+    });
+
+    test('it returns false if a simple string value does not match', () => {
+      expect(isIgnored('simple', ['simple.value'])).toEqual(false);
+    });
+
+    test('it returns true if a simple string value matches with two strings', () => {
+      expect(isIgnored('simple.value', ['simple.value', 'simple.second.value'])).toEqual(true);
+    });
+
+    test('it returns true if a simple string value matches the second string', () => {
+      expect(isIgnored('simple.second.value', ['simple.value', 'simple.second.value'])).toEqual(
+        true
+      );
+    });
+
+    test('it returns false if a simple string value does not match two strings', () => {
+      expect(isIgnored('simple', ['simple.value', 'simple.second.value'])).toEqual(false);
+    });
+
+    test('it returns true if mixed with a regular expression in the list', () => {
+      expect(isIgnored('simple', ['simple', '/[_]+/'])).toEqual(true);
+    });
+  });
+
+  describe('regular expression matching', () => {
```

Comment:
Might be worth adding a test for an invalid regular expression.

Reply:
Tests for the user validation input are here:
https://github.com/elastic/kibana/pull/110927/commits/e755c3ed24d0ef5bf6648c7fa6cde8344d68d93e